src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -0,0 +1,309 @@
+    ///  
+    /// Current activity can be accessed with static AsyncLocal variable Activity.Current.
+    /// 
+    /// Activities should be created with constructor, configured as necessarily 
+    /// and then started with Activity.Start method which maintaines parent-child
+    /// relationships for the activities and sets Activity.Current.
+    /// 
+    /// When activity is finished, it should be stopped with static Activity.Stop method.
+    /// </summary>
+    public partial class Activity
+    {
+        /// <summary>
+        /// An operation name is a COARSEST name that is useful grouping/filtering. 
+        /// The name is typically a compile time constant.   Names of Rest APIs are 
+        /// reasonable, but arguments (e.g. specific accounts etc), should not be in
+        /// the name but rather in the tags.  
+        /// </summary>
+        public string OperationName { get; }
+
+        /// <summary>
+        /// This is an ID that is specific to a particular request.   Filtering
+        /// to a particular ID insures that you get only one request that matches.  
+        /// It is typically assigned the system itself. 
+        /// </summary>
+        public string Id { get; private set; }
+
+        /// <summary>
+        /// The time that operation started.  Typcially when Start() is called 
+        /// (but you can pass a value to Start() if necessary.  This use UTC (Greenwitch Mean Time)
+        /// </summary>
+        public DateTime StartTimeUtc { get; private set; }
+
+        /// <summary>
+        /// If the Activity that created this activity is  from the same process you can get 
+        /// that Activit with Parent.   However this can be null if the Activity has no
+        /// parent (a root activity) or if the Parent is from outside the process.  (see ParentId for more)
+        /// </summary>
+        public Activity Parent { get; private set; }
+
+        /// <summary>
+        /// If the parent for this activity comes from outside the process, the activity
+        /// does not have a Parent Activity but MAY have a ParentId (which was serialized from
+        /// from the parent) .   This accessor fetches the parent ID if it exists at all.  
+        /// Note this can be null if this is a root Activity (it has no parent)
+        /// </summary>
+        public string ParentId { get; private set; }
+
+        /// <summary>
+        /// Tags are string-string key-value pairs that represent information that will
+        /// be logged along with the Activity to the logging system.   This information
+        /// however is NOT passed on to the children of this activity.  (see Baggage)
+        /// </summary>
+        public IEnumerable<KeyValuePair<string, string>> Tags
+        {
+            get
+            {
+                if (ParentId != null)
+                    yield return new KeyValuePair<string, string>("ParentId", ParentId);
```

Comment:
ParentId is returned in Tags to simplify activity logging: we include ParentId to make sure it will be logged. It's good idea to apply the same behavior to Id and operation name.